### PR TITLE
Feat: switch from essentia to beatnet

### PR DESCRIPTION
**Pull request by faradox**
We found BeatNet to run faster and produce better results than essentia. This PR introduces BeatNet instead of essentia for quantization.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ nendo = "^0.1.2"
 pydantic = { version = "^2.0.0, <2.5.0" }
 pydantic-settings = "^2.1.0"
 pyrubberband = "^0.3.0"
-essentia = "^2.1b6.dev1110"
+BeatNet = "^1.1.3"
```
Comment (aaronabebe) [inline pyproject.toml:44]:
```suggestion
BeatNet = "^1.1.3"
PyAudio = "^0.2.14"
```

From the BeatNet repo: 
> Note: Before installing the BeatNet make sure [Librosa](https://librosa.org/) and [Madmom](https://madmom.readthedocs.io/en/latest/installation.html) packages are installed. Also, pyaudio is a python binding for Portaudio to handle audio streaming. If Pyaudio is not installed in your machine, depending on your machine type either install it thorugh pip (Mac OS and Linux) or download an appropriate version for your machine (Windows) from [here](https://www.lfd.uci.edu/~gohlke/pythonlibs/). Then, navigate to the file location through commandline and use the following command to install the wheel file locally:

File: src/nendo_plugin_quantize_core/plugin.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
 """A nendo core plugin for music quantization."""
 import math
 from logging import Logger
-from typing import List, Tuple
+from typing import List
 
 import essentia.standard as es
```
Comment (aaronabebe) [inline src/nendo_plugin_quantize_core/plugin.py]:
```suggestion
```

File: src/nendo_plugin_quantize_core/plugin.py [hunk 3]
```diff
@@ -75,32 +76,69 @@ def extract_beat(self, y, sr):
...
+
+        # Average interval
+        avg_interval = sum(intervals) / len(intervals)
+
+        # Convert to BPM (60 seconds per minute)
+        tempo = 60 / avg_interval
+
+        return tempo, beat_frames
+
+    # def extract_beat_essentia(self, y, sr):
```
Comment (aaronabebe) [inline src/nendo_plugin_quantize_core/plugin.py]: any reason to not remove this instead of just commenting it out?

**Review by aaronabebe — CHANGES_REQUESTED** [review]

File: src/nendo_plugin_quantize_core/plugin.py (hunk 3 above)
Comment (faradox) [reply src/nendo_plugin_quantize_core/plugin.py]: good point. will remove

**Review by aaronabebe — APPROVED** [review]

**Pull request merged**